pytket/extensions/qiskit/backends/ibm.py

```diff
@@ -120,6 +124,18 @@ def __init__(self) -> None:
         )
 
 
+class GateSet(Enum):
+    X_SX_RZ_CX = {OpType.X, OpType.SX, OpType.Rz, OpType.CX}
+    X_SX_RZ_ECR = {OpType.X, OpType.SX, OpType.Rz, OpType.ECR}
```

Comment:
I don't think we need this enum at all. We are storing the primitive gates as a backend attribute -- so just use that attribute when computing `rebase_pass()`?

Reply:
Done. 

I've gotten rid of the enum and used the backend attribute when possible.

For the `_get_primitive_gates` function I've used

```python
def _get_primitive_gates(gateset: Set[OpType]) -> Set[OpType]:
    if gateset >= {OpType.X, OpType.SX, OpType.Rz, OpType.CX}:
        return {OpType.X, OpType.SX, OpType.Rz, OpType.CX}
    elif gateset >= {OpType.X, OpType.SX, OpType.Rz, OpType.ECR}:
        return {OpType.X, OpType.SX, OpType.Rz, OpType.ECR}
    else:
        return gateset
```